messages/message_handler.go

```diff
@@ -22,11 +23,11 @@ type MessageHandlerFactory interface {
 type MessageHandler interface {
 	// ShouldSendMessage returns true if the message should be sent to the destination chain
 	// If an error is returned, the boolean should be ignored by the caller.
-	ShouldSendMessage(destinationBlockchainID ids.ID) (bool, error)
+	ShouldSendMessage(destinationClient vms.DestinationClient) (bool, error)
 
 	// SendMessage sends the signed message to the destination chain. The payload parsed according to
 	// the VM rules is also passed in, since MessageManager does not assume any particular VM
-	SendMessage(signedMessage *warp.Message, destinationBlockchainID ids.ID) error
+	SendMessage(signedMessage *warp.Message, destinationClient vms.DestinationClient) error
```

Comment:
Now that the destination client is a field in the message handler, we don't need to pass it into these functions anymore

Reply:
The destination client is a field in the application relayer, not the message handler. The two objects are decoupled (see how they're fetched for a particular message in `Listener.GetAppRelayerMessageHandler`) and it would take some refactoring to couple them more closely. I'm not convinced that they should be coupled. In any case, I'd prefer to defer any more refactoring that's not strictly related to concurrent message processing to a later change.

Reply:
Oh, got it.